doc/y_cable/design_doc.md

```diff
@@ -0,0 +1,226 @@
+## Multiple y cable vendor API design
+
```

Comment:
Suggest adding a "Revision" table ([example](https://github.com/Azure/SONiC/blob/master/doc/system-telemetry/reboot-cause.md#revision)) in case changes are made in the future, so we can keep track.

Reply:
fixed